src/Message/Handler/AssignProductCalloutsHandler.php

```diff
@@ -33,14 +34,9 @@ public function __construct(
         $this->productManager = $productManager;
     }
 
-    public function execute(AMQPMessage $message): void
+    public function __invoke(AssignProductCallouts $message)
     {
-        $body = unserialize($message->getBody());
-
-        Assert::isArray($body);
-        Assert::keyExists($body, 'products');
-
-        $products = $this->productRepository->findBy(['id' => $body['products']]);
+        $products = $this->productRepository->findBy(['id' => $message->getProductIds()]);
```

Comment:
Does this work? Does Doctrine convert this to a `IN()` expression behind the scenes? :)

Reply:
yes